packages/ckeditor5-restricted-editing/src/restrictededitingmode/converters.ts

```diff
@@ -106,9 +106,10 @@ export function extendMarkerOnTypingPostFixer( editor: Editor ): ModelPostFixer
 	// This post-fixer shouldn't be necessary after https://github.com/ckeditor/ckeditor5/issues/5778.
 	return writer => {
 		let changeApplied = false;
+		const schema = editor.model.schema;
 
 		for ( const change of editor.model.document.differ.getChanges() ) {
-			if ( change.type == 'insert' && change.name == '$text' ) {
+			if ( change.type == 'insert' && schema.checkChild( '$block', change.name ) ) {
```

Comment:
Why does this work? Are there other cases that are handled in addition to the inline image? Asking because I'm not that familiar with schema and want to learn something new 🙂

Reply:
Our schema registers relation between various types of elements in editor. For example `$block` elements e.g. `paragraph` allows `$text` inside or other inline elements. `schema.checkChild(...)` verifies does element allowed be inside `$block` or not.